### PR TITLE
Added iGrowl w/ git auto-update, closes #10205

### LINKED ISSUE #10205: [Request] Add iGrowl
**Library name:** iGrowl
**Git repository url:** https://github.com/catc/iGrowl
**License(s):** MIT
**Official homepage:** https://catc.github.io/iGrowl/

For beginners who is interested in picking up this issue:

In this issue, we are going to host a new web front-end library on CDNJS, so that web developer can easily use it on our free CDN without additional download/upload process 🚀 .

If you'll like to work locally, I'll suggest you start with sparseCheckout: https://github.com/cdnjs/cdnjs/blob/master/documents/sparseCheckout.md, since there are too many files in this repo, without sparseCheckout, the whole process will be super slow, using git command line can learn more skills and experience, if you don't like that, instead, you can also just do it on GitHub, it's more convenient, but it may be a little bit harder to fix problems on GitHub gui.

Here are the docs about how to add a library:
 - https://github.com/cdnjs/cdnjs#adding-a-new-or-updating-an-existing-library
 - https://github.com/cdnjs/cdnjs/blob/master/.github/CONTRIBUTING.md#d-adding-a-new-library

We'll also ask to setup auto-update config so that the added libraries can be updated to the latest version easily:
 - https://github.com/cdnjs/cdnjs/blob/master/documents/autoupdate.md

For beginners who is interested in join and help add this library, please feel free to tell me what kind of help you'll need, we'll help you send the valid pull request, thanks 😄 .

### PR DESCRIPTION
**Pull request by elhadiahmed**
Pull request for issue: #10205
Related issue(s): # #

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/catc/iGrowl
 * Official website (optional, not the repository): https://catc.github.io/iGrowl/
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 6
   - Count of stars: 207
   - Count of forks: 20
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [x] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [ ] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [ ] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [ ] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [ ] Separate unrelated changes into different commits.
 * [ ] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [ ] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by elhadiahmed** [comment]
@PeterDaveHello sorry, what property is required ?

```
    iGrowl 
      ✗ iGrowl package.json is valid 
        » iGrowl is not a valid cdnjs package.json format: 
  	  » NPM package.json 
  		Property is required (true): /properties/filename // valid-packages-test.js:87 
  ✗ Broken » 28 honored ∙ 1 broken (0.022s) 
  npm ERR! Test failed.  See above for more details.

==========ERROR==========
npm test failed!
==========ERROR==========
```

**Comment by PeterDaveHello** [comment]
@elhadiahmed you need "filename" to point to the main file, e.g.
https://github.com/cdnjs/cdnjs/blob/master/ajax/libs/jquery/package.json#L12
https://github.com/cdnjs/cdnjs/tree/master/ajax/libs/jquery/3.1.1

**Comment by elhadiahmed** [comment]
@PeterDaveHello the main js file ? how about the css should it be listed as an array ?

**Comment by PeterDaveHello** [comment]
@elhadiahmed yes, main js or main css file, only one main file, no array

File: ajax/libs/iGrowl/package.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	"author": {
+		"name": "Catalin Covic",
+		"email": "catalin.covic@gmail.com",
+		"url": "https://devpost.com/catcc"
+	},
+	"repository": {
+		"type": "git",
+		"url": "https://github.com/catc/iGrowl"
+	},
+	"keywords": [
```
Comment (pvnr0082t) [inline ajax/libs/iGrowl/package.json]: @elhadiahmed There are more keywords from upstream. Please use these https://github.com/catc/iGrowl/blob/3.0.3/bower.json#L10-L17. Thanks.

File: ajax/libs/iGrowl/package.json [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+		"alerts"
+	],
+	"license": "MIT",
+	"autoupdate": {
+		"source": "git",
+		"target": "git://github.com/catc/iGrowl.git",
+		"fileMap": [{
+			"basePath": "dist",
+			"files": [
+				"**/*"
```
Comment (pvnr0082t) [inline ajax/libs/iGrowl/package.json]: Because it will fetch `*.json` files by using `"**/*"`, let's use `"**/*.!(json)"` instead.

**Comment by pvnr0082t** [comment]
@PeterDaveHello Should we care about the indent of package.json?

**Comment by PeterDaveHello** [comment]
No worry about indent for single package.json library adding process.

File: ajax/libs/iGrowl/package.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{
+  "name": "iGrowl",
+  "filename": "igrowl.min.js",
```
Comment (PeterDaveHello) [inline ajax/libs/iGrowl/package.json]: It should be `js/igrowl.min.js`

Comment (elhadiahmed) [inline ajax/libs/iGrowl/package.json]: ok

**Pull request merged**